### PR TITLE
feat: rate limiting

### PR DESCRIPTION
**Pull request by chris13524**
# Description

Adds rate limiting.

- [x] Re-enable other SIWE statement tests
- [x] Refactor and add limits to subscriber notifications
- [x] Test subscriber rate limit & multiple subscribers
- [x] Test responses in detail: not subscribed, not subscribed to scope, subscribed but rate limited
- [ ] ~~Attempt refactoring of notify endpoint to reduce complexity~~
  - Couldn't think of an obvious way to simplify
- [x] Add rate limits to other endpoints
- [ ] ~~Rate limit data export for each endpoint call, containing the project and remaining tokens~~
  - Can do in follow-up PR

Resolves https://github.com/WalletConnect/notify-server/issues/14

## How Has This Been Tested?

Automated tests

## Due Diligence

* [ ] Breaking change
* [ ] Requires a documentation update
* [ ] Requires a e2e/integration test update

### CODE REVIEW

File: src/auth.rs [hunk 1]
```diff
@@ -539,6 +525,24 @@ pub async fn verify_identity(
     })
 }
 
+fn parse_cacao_statement(statement: &str, domain: &str) -> Result<AuthorizedApp> {
```
Comment (geekbrother) [inline src/auth.rs:528]: should be in a different PR as it seems is not related to the rate limiting

Comment (chris13524) [reply src/auth.rs:528]: It's related because the rate limit prevents more than 5 `/notify` calls in a single second, and the tests all use the same project ID. So had to refactor the type of tests to allow them to pass.

**Review by geekbrother — APPROVED** [review]
Looks good and I trust your tests in `integration.rs`.
